composer/datasets/streaming/format.py

```diff
@@ -0,0 +1,237 @@
+import math
+from io import BufferedIOBase, BufferedReader, BufferedWriter, BytesIO
+from typing import Dict, Optional, Sequence, Tuple, Union
+
+import numpy as np
+from numpy.typing import DTypeLike, NDArray
+
+from composer.datasets.streaming.world import World
+
+
+def get_index_basename() -> str:
+    """Get the basename for a streaming dataset index.
+
+    Returns:
+        str: Basename of file.
+    """
+    return 'index.mds'
+
+
+def get_shard_basename(shard: int) -> str:
+    """Get the basename for a streaming dataset shard.
+
+    Args:
+        shard (int): Which shard.
+
+    Returns:
+        str: Basename of file.
+    """
+    return f'{shard:05}.mds'
```

Comment:
I think this is an issue with shard writing, we don't know the number of shards ahead of time as the samples are arbitrary size and we write out shards with an approx `shard_size_limit` in _byes,_ not samples.

Luckily this will not produce a bug, it will just be slightly inconsistent filenames if the `num_shards` exceeds 10^5 (or whatever value we choose):

```
>>> print (f"{123:05}")
00123
>>> print (f"{123456:05}")
123456
```

I will increase the zero buffer to 6 so that it's up to 100,000 shards... I think that should keep us safe for now. I will track this issue as well so we can implement a better fix.